### PR TITLE
Improve UX in redemption details page

### PR DESCRIPTION
**Pull request by r-czajkowski**
Redirect a user to a success step automatically when the redemption is complete. Here we subscribe to the `RedemptionsCompleted` event and check if this event is related to a given redemption request by checking the wallet public key hash. If yes, we are looking for a Bitcoin transfer to a given redeemer output script in a given bitcoin tx hash (we can get this transaction hash from the `RedemptionsCompleted` event). If we are able to find that transfer it means this `RedemptionsCompleted` event is related to a given redemption request and we should redirect user to a success step and update the bridge activity list.

### CODE REVIEW

File: src/hooks/tbtc/useSubscribeToRedemptionsCompletedEvent.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import { useSubscribeToContractEvent } from "../../web3/hooks"
+import { useBridgeContract } from "./useBridgeContract"
+
+type RedemptionsCompletedEventCallback = (
+  walletPublicKeyHash: string,
+  redemptionTxHash: string,
+  event: Event
+) => void
+
+export const useSubscribeToRedemptionsCompletedEventBase = (
```
Comment (michalsmiarowski) [inline src/hooks/tbtc/useSubscribeToRedemptionsCompletedEvent.ts:12]: Can you explain why you put base word at the end of the hook name? Wouldn't `useSubscribeToRedemptionsCompletedEvent` be enough?

**Review by michalsmiarowski — CHANGES_REQUESTED** [review]
Overall looking very good. Left small non-blocking comments to look at before the merge

File: src/hooks/tbtc/useSubscribeToRedemptionsCompletedEvent.ts (hunk 1 above)
Comment (r-czajkowski) [reply src/hooks/tbtc/useSubscribeToRedemptionsCompletedEvent.ts:12]: We usually use `useSubscribeToXEvent` when updating the redux store based on the events. This hook is to share a common logic where we reverse the `redemptionTxHash` param to represent the Bitcoin transaction hash in the same byte order used by the Bitcoin block explorers. That's why I use `base` word at the end. Eg https://github.com/threshold-network/token-dashboard/blob/main/src/hooks/tbtc/useSubscribeToOptimisticMintingRequestedEvent.ts#L19

File: src/pages/tBTC/Bridge/UnmintDetails.tsx [hunk 2]
```diff
@@ -54,39 +54,84 @@ import { featureFlags } from "../../../constants"
...
 
   const _isFetching = (isFetching || !data) && !error
   const wasDataFetched = !isFetching && !!data && !error
 
-  const btcTxHash = data?.redemptionCompletedTxHash?.bitcoin
-  useEffect(() => {
-    setShouldDisplaySuccessStep(!!btcTxHash)
-  }, [btcTxHash])
+  const isProcessCompleted = !!redemptionFromBitcoinTx?.bitcoinTxHash
+  const shoudlForceIsProcessCompleted =
```
[the remark this answers is not in the document]
Comment (r-czajkowski) [reply src/pages/tBTC/Bridge/UnmintDetails.tsx]: 37e6ef3

File: src/hooks/tbtc/useFetchRedemptionDetails.ts [hunk 3]
```diff
@@ -194,44 +194,34 @@ export const useFetchRedemptionDetails = (
...
+              chain: txHash,
+              bitcoin: redemptionBitcoinTxHash,
+            },
+            requestedAt: redemptionRequestedEventTimestamp,
+            completedAt: redemptionBitcoinTransfer.redemptionCompletedTimestamp,
+            treasuryFee: fromSatoshiToTokenPrecision(
+              redemptionRequestedEvent.treasuryFee
+            ).toString(),
+            isTimedOut: false,
+            btcAddress: redemptionBitcoinTransfer.btcAddress,
```
[the remark this answers is not in the document]
Comment (r-czajkowski) [reply src/hooks/tbtc/useFetchRedemptionDetails.ts]: 1881695

**Review by michalsmiarowski — APPROVED** [review]
LGTM 🔥

**Pull request merged**